### PR TITLE
CI: Enhance error handling in Azure Pipelines

### PR DESCRIPTION
**Pull request by morkev**
Enhance error handling in Azure Pipelines by adding verbose logging, implementing `continueOnError` for non-critical steps, and improving condition checks and alerts for more robust CI/CD operations.

### CODE REVIEW

**Review by morkev — COMMENTED** [review]
Ensures that all 'git' submodules are fetched and initialized before the build and test steps commence, this update guarantees that all necessary code dependencies are present and up-to-date.

**Comment by morkev** [comment]
Hi @charris . Any chance this can get merged, or should we just close the PR? Thank you so much.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
       - main
       - maintenance/*
 
-
 pr:
   branches:
     include:
-    - '*'  # must quote since "*" is a YAML reserved character; we want a string
-
+      - '*'  # All branches for PRs
```
Comment (mattip) [inline azure-pipelines.yml:11]: Unindent?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -50,8 +46,7 @@ stages:
     - script: >-
         python -m pip install -r requirements/linter_requirements.txt
       displayName: 'Install tools'
-      # pip 21.1 emits a pile of garbage messages to annoy users :)
-      #      failOnStderr: true
+      continueOnError: true
```
Comment (mattip) [inline azure-pipelines.yml:49]: I don't understand. How can the lint job succeed if the tools are not installed?

File: azure-pipelines.yml [hunk 3]
```diff
@@ -65,11 +60,11 @@ stages:
         git submodule update --init
       displayName: 'Fetch submodules'
     - script: |
-            # yum does not have a ninja package, so use the PyPI one
             docker run -v $(pwd):/numpy -e CFLAGS="-msse2 -std=c99 -UNDEBUG" \
             -e F77=gfortran-5 -e F90=gfortran-5 quay.io/pypa/manylinux2014_i686 \
             /bin/bash -xc "source /numpy/tools/ci/run_32_bit_linux_docker.sh"
       displayName: 'Run 32-bit manylinux2014 Docker Build / Tests'
+      continueOnError: true  # Consider whether to allow continuation on error
```
Comment (mattip) [inline azure-pipelines.yml:67]: This seems wrong. If the tests fail, the job should fail

**Review by mattip — COMMENTED** [review]
The changes to azure-steps-windows.yml are not needed. This is a template, it is injected into the azure-pipelines.yml file so should not have a `trigger` and `stages` stanza.


Is there some deeper motivation for this PR: is there an issue or problem it comes to solve?

**Comment by mattip** [comment]
@morkev will you be continuing with this?

**Comment by morkev** [comment]
@mattip Hi, thanks for tagging me. As you mentioned earlier, this PR does solve any particular issue, and I would like to contribute with something better. For the exception part, I agree - why catch an error if you can’t handle it. The indentation was probably a mistake from the IDE, would only change if it’s better based on PEP 8.

**Pull request closed without merging**